test/util/splitPath-spec.js

```diff
@@ -0,0 +1,25 @@
+import { expect } from 'chai';
+import splitPath from '../../lib/util/splitPath';
+
+describe('splitPath', () => {
+  it('test number path', () => {
+    const path = 1;
+    const result = splitPath(path);
+    expect(result).to.deep.equal(['1']);
+  });
+  it('test number path', () => {
+    const path = 0;
+    const result = splitPath(path);
+    expect(result).to.deep.equal(['0']);
+  });
+  it('test array path', () => {
+    const path = ['foo', 'bar', 'x'];
+    const result = splitPath(path);
+    expect(result).to.equal(path);
+  });
+  it('test string path', () => {
+    const path = 'bar.0.y';
+    const result = splitPath(path);
+    expect(result).to.deep.equal(['bar', '0', 'y']);
+  });
```

Comment:
sorry, my example was meant to be for the first.

You can certainly simplify, but I would use the word "handles" instead of "treats", since "treats" requires something like "as a ..." at the end of the sentence.

So, "it handles numbers", "it handles arrays", "it handles strings separated by dots" 

You can probably get rid of either the 0 or the 1 test, you don't really need them both and they'd have the same name.